packages/http-server/src/http-server.ts

```diff
@@ -35,42 +41,55 @@ export class HttpServer {
   private _listening: boolean = false;
   private _protocol: HttpProtocol;
   private _address: AddressInfo;
-  private httpRequestListener: HttpRequestListener;
-  private httpServer: Server;
+  private requestListener: RequestListener;
+  private server: Server | HttpsServer;
+  private serverOptions?: HttpServerOptions;
 
   /**
-   * @param httpServerOptions
-   * @param httpRequestListener
+   * @param requestListener
+   * @param serverOptions
    */
   constructor(
-    httpRequestListener: HttpRequestListener,
-    httpServerOptions?: HttpServerOptions,
+    requestListener: RequestListener,
+    serverOptions?: HttpServerOptions,
   ) {
-    this.httpRequestListener = httpRequestListener;
-    if (!httpServerOptions) httpServerOptions = {};
-    this._port = httpServerOptions.port || 0;
-    this._host = httpServerOptions.host || undefined;
-    this._protocol = httpServerOptions.protocol || 'http';
+    this.requestListener = requestListener;
+    this.serverOptions = serverOptions;
+    this._port = serverOptions ? serverOptions.port || 0 : 0;
+    this._host = serverOptions ? serverOptions.host : undefined;
+    this._protocol = serverOptions ? serverOptions.protocol || 'http' : 'http';
   }
 
   /**
    * Starts the HTTP / HTTPS server
    */
   public async start() {
-    this.httpServer = createServer(this.httpRequestListener);
-    this.httpServer.listen(this._port, this._host);
-    await pEvent(this.httpServer, 'listening');
-    this._listening = true;
-    this._address = this.httpServer.address() as AddressInfo;
+    if (this._protocol === 'https') {
+      const httpsOptions = Object.assign({}, this.serverOptions);
```

Comment:
Is it necessary to create a copy of the options object? Can we pass `this.serverOptions` to `https.createServer` directly?